docs/architecture/adr-028-public-key-addresses.md

```diff
@@ -56,42 +89,90 @@ The current multisig public keys use amino serialization to generate the address
 those public keys and their address formatting, and call them "legacy amino" multisig public keys
 in protobuf. We will also create multisig public keys without amino addresses to be described below.
 
+### Hash Function Choice
+
+We propose to use [blake2b](https://www.blake2.net/) as a hash function choice:
++ The main arguments are speed and separating from `sha256` which is widely used
+  by miners and could potentially be used to find collisions.
++ The function was in the final round of the 2012 NIST hash function competition.
++ It's well studied with security covered in many academic papers.
++ Faster than `sha2` on non ASICs chipsets.
++ It's getting more traction in other blockchains (Pokadot, Sia, Zcash, ...). Related [zcash discussion](https://github.com/zcash/zcash/issues/706#issuecomment-187807410).
++ It's already widely supported by all major programming languages.
++ Cryptography consulting reviled no argument against `blake2b`
+
+
+### Base Address Algorithm
+
+We start with defining a base algorithm for generating addresses. Notably, it's used for Base Accounts (accounts represented by a single key-pair) addresses. For each Public Key schema we need to have an associated `typ` string, which we will discuss in a section below. `hash` is a cryptographic hash function defined in the previous section.
+
+```go
+const A_LEN = 32
+
+func BaseAddress(typ string, pubkey []byte) []byte {
+    return hash(hash(typ) + pubkey)[:A_LEN]
```

Comment:
Why not do a personalization string if the underlying hash supports it? 

Its also worth noting, that for efficiency, `hash(typ)` should ideally have length `block size` OR `size(hash(type) + pubkey) < 1 block size`

Reply:
I'm not sure I follow you:
* What do you mean with _ personalization string_?
* The returned bytes are capped at `A_LEN`. `hash` is defined in the section above. I specifically, didn't want to name it here -- in case we will edit it, we only need to change one place. 
* With `block size` - you mane a blockchain block size? That's totally different scale.